### ISSUE TITLE
Every 4 days AdGuard Home Stop Working

### ISSUE THREAD

**Issue by Adgbeta** [AdguardTeam/AdGuardHome#6488]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

FreeBSD, AMD64 (aka x86_64)

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

0.108.0-b.50

### Action

Every 4 days AdGuard Home Stop Working, not internet access for the whole network, DNS stop working. After re-start the machine, internet access start again without issues, for 4 days then DNS stop working. Since November 22, I had to re-start 2 times.

### Expected result

I expect AdGuard Home work without interruptions and not need to re-boot every 4 days

### Actual result

Need to re-boot every 4 days, to have internet access.

### Additional information and/or screenshots

Router pfsense, using Kea DHCP.

**Comment by agneevX** [comment]
Yeah I can attest to this. I've downgraded to .39 to resolve this but I get random memory leaks and high CPU from time to time.

**Comment by CyrusHou** [comment]
Same situation as yours

**Comment by CyrusHou** [comment]
以收到,谢谢

**Comment by Adgbeta** [comment]
Testing edge build, so far no increase of memory use, not crash DNS server.

**Comment by coolxy** [comment]
Yes, my device is the same, and I think there is a problem with its storage. I have replaced it with a new one, but the problem still exists. Later, it was changed to version 107.41.

**Comment by jzm2bjo** [comment]
Same on both MacOS native and windows server 22 x64

**Comment by CyrusHou** [comment]
以收到,谢谢

**Comment by ainar-g** [comment]
@Adgbeta, is the beta still working well for you?

**Comment by CyrusHou** [comment]
以收到,谢谢

**Comment by ivicask** [comment]
Same issue it just stops working every few days runing in mikrotik router container , happens on stable and latest edge builds past few months of releases .

**Comment by VK3002** [comment]
The same issue. Happens on my PC(Windows10, x86_64). AdGuard Home v0.107.44.
It seems have something to do with memory problem because It had no respond on port 53 and got stuck with high memory occupation(this maybe just caused by memory leaking, I'm not sure).

**Comment by CyrusHou** [comment]
以收到,谢谢

**Comment by DrMerfy** [comment]
Same issue seems to appear for both AdGuard Home latest and edge running on container inside Mikrotik hap ac2. Could it be related to small RAM size (128MB)?

>Jun/23/2024 10:06:44 system,error,critical router was rebooted without proper shutdown, probably kernel failure
Jun/23/2024 10:06:46 system,error,critical kernel failure in previous boot
Jun/23/2024 10:06:46 system,error,critical out of memory condition was detected
Jun/23/2024 10:18:11 system,error,critical router was rebooted without proper shutdown, probably kernel failure
Jun/23/2024 10:18:12 system,error,critical kernel failure in previous boot
Jun/23/2024 10:18:12 system,error,critical out of memory condition was detected
Jun/25/2024 10:18:25 system,error,critical router was rebooted without proper shutdown, probably kernel failure
Jun/25/2024 10:18:26 system,error,critical kernel failure in previous boot
Jun/25/2024 10:18:26 system,error,critical out of memory condition was detected
Jun/26/2024 10:22:44 system,error,critical router was rebooted without proper shutdown, probably kernel failure
Jun/26/2024 10:22:46 system,error,critical kernel failure in previous boot
Jun/26/2024 10:22:46 system,error,critical out of memory condition was detected
Jun/26/2024 10:31:36 system,error,critical router was rebooted without proper shutdown, probably kernel failure
Jun/26/2024 10:31:36 system,error,critical kernel failure in previous boot
Jun/26/2024 10:31:36 system,error,critical out of memory condition was detected
Jun/27/2024 10:33:21 system,error,critical router was rebooted without proper shutdown, probably kernel failure
Jun/27/2024 10:33:22 system,error,critical kernel failure in previous boot
Jun/27/2024 10:33:22 system,error,critical out of memory condition was detected
Jun/29/2024 10:32:57 system,error,critical router was rebooted without proper shutdown, probably kernel failure
Jun/29/2024 10:32:59 system,error,critical kernel failure in previous boot
Jun/29/2024 10:32:59 system,error,critical out of memory condition was detected

![image](https://github.com/AdguardTeam/AdGuardHome/assets/21154979/1add08f6-de83-4bdb-b2d7-57c1172cae27)

**Comment by CyrusHou** [comment]
以收到,谢谢
